src/test/java/bio/terra/integration/DataRepoFixtures.java

```diff
@@ -687,6 +691,50 @@ public SnapshotModel getSnapshot(
     return validateResponse(response, "snapshot retrieve", HttpStatus.OK, null);
   }
 
+  /**
+   * Given a dataset, table, and column, retrieve the DRS URI from the snapshot preview and extract
+   * its DRS object ID.
+   *
+   * <p>In the past, we've queried BigQuery directly for this, but unpredictable delays in IAM
+   * propagation make going through the preview more reliable (TDR proxies its request to BigQuery
+   * here).
+   */
+  public String retrieveDrsIdFromSnapshotPreview(
+      TestConfiguration.User user, UUID snapshotId, String table, String column) throws Exception {
+    String filter = "WHERE %s IS NOT NULL".formatted(column);
+    DataRepoResponse<SnapshotPreviewModel> response =
+        retrieveSnapshotPreviewByIdRaw(user, snapshotId, table, 0, 1, filter);
+    SnapshotPreviewModel validated =
+        validateResponse(response, "snapshot preview for DRS ID", HttpStatus.OK, null);
+
+    assertThat("Got one row", validated.getResult(), hasSize(1));
+
+    String drsUri = (String) ((LinkedHashMap) validated.getResult().get(0)).get(column);
+    assertThat("DRS URI was found", drsUri, notNullValue());
+
+    return DrsService.getLastNameFromPath(drsUri);
```

Comment:
Is this to specific to the test to be in the fixture?  (e.g. feels like we should just return the drs id (if anything, we could validate that the string is a valid drs id).  If we return a drs id, we could actually return it as a `DrsId` object